core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2509,12 +2509,10 @@ class KafkaApis(val requestChannel: RequestChannel,
             .setThrottleTimeMs(requestThrottleMs)))
       case Some(auth) =>
         val filter = describeAclsRequest.filter
-        val returnedAcls = new util.HashSet[AclBinding]()
-        auth.acls(filter).forEach(returnedAcls.add)
```

Comment:
Note that this means we no longer filter duplicates. We should make sure that doesn't cause issues.

Reply:
Good point! It seems to me the method ```aclsResources``` should de-duplicate as it tries to traverse all elements from input